engine/engine.go

```diff
@@ -69,6 +69,11 @@ func (e *Engine) Get(m multihash.Multihash) ([]indexer.Value, bool, error) {
 	return e.valueStore.Get(m)
 }
 
+// ForEach iterates multihashes in the value store, calling the provided function for each
+func (e *Engine) ForEach(iterFunc indexer.IterFunc) error {
```

Comment:
I personally prefer a `ForEach` for the iteration API than an `Iteration + Next/Done` approach, it is more compact and explicit, but let's double-check with Ignite if this suits their needs. I'll ping them right now.

Reply:
//cc @aarshkshah1992

Reply:
It would be trivial to provide this by having an `Iter()` method that returns an `Iterator` interface.  Given that I could see no advantage between ForEach and Iterator object approach, I chose ForEach since that does not require introducing a new type (however trivial) for each value store.

If an iterator object makes it easier to use, I will be happy to change this.